diff --git a/ui/src/controls.rs b/ui/src/controls.rs
index 5699276..56d1c1e 100644
--- a/ui/src/controls.rs
+++ b/ui/src/controls.rs
@@ -242,7 +242,7 @@ impl Button {
                 let button = Button {
                     ui_button: button,
                 };
-                mem::transmute::<*mut c_void, &mut Box<FnMut(&Button)>>(data)(&button)
+                &mut *(data as *mut Box<FnMut(&Button)>)(&button)
             }
         }
     }
