diff --git a/ui/src/controls.rs b/ui/src/controls.rs
index 5699276..59e36b9 100644
--- a/ui/src/controls.rs
+++ b/ui/src/controls.rs
@@ -242,7 +242,7 @@ impl Button {
                 let button = Button {
                     ui_button: button,
                 };
-                mem::transmute::<*mut c_void, &mut Box<FnMut(&Button)>>(data)(&button)
+                &mut *(data as *mut Box<FnMut(&Button)>)(&button)
             }
         }
     }
@@ -347,7 +347,7 @@ impl Entry {
         extern "C" fn c_callback(entry: *mut uiEntry, data: *mut c_void) {
             unsafe {
                 let entry = Entry::from_ui_control(entry);
-                mem::transmute::<*mut c_void, &mut Box<FnMut(&Entry)>>(data)(&entry);
+                &mut *(data as *mut Box<FnMut(&Entry)>)(&entry);
                 mem::forget(entry);
             }
         }
